### DISCUSSION TITLE
HIP report feature

### DISCUSSION THREAD

**Discussion by filipmestrovic** [yuezk/GlobalProtect-openconnect#323] [General]
After connecting I see a log like this:
```
WARN  openconnect::ffi] WARNING: Server asked us to submit HIP report with md5sum [x].
        VPN connectivity may be disabled or limited without HIP report submission.
        You need to provide a --csd-wrapper argument with the HIP report submission script
```

Is the new HIP support feature send any reports?
Can it also send information about installed antivirus software?

Thanks!

**Comment by yuezk** [comment]
If you check the `Submit HIP Report` option, it will use the default CSD wrapper script to generate and submit the report.

The default script is https://gitlab.com/openconnect/openconnect/-/blob/master/trojans/hipreport.sh?ref_type=heads, it will generate antivirus only if the client os is `Windows`, which can be configured on the same page.

<img width="611" alt="image" src="https://github.com/yuezk/GlobalProtect-openconnect/assets/3297602/b458ff99-b469-4466-823d-fd93aa25a3e6">

**Reply by NoahDavon** [reply]
Any idea how to do that on the cli version?

**Reply by yuezk** [reply]
@NoahDavon You use the `--hip` parameter with the CLI version. See `gpclient connect --help` for more information.
